strawberryfields/api/connection.py

```diff
@@ -301,12 +305,43 @@ def ping(self) -> bool:
             bool: ``True`` if the connection is successful, and ``False`` otherwise
         """
         path = "/healthz"
-        response = requests.get(self._url(path), headers=self._headers)
+        response = self._request("GET", self._url(path), headers=self._headers)
         return response.status_code == 200
 
     def _url(self, path: str) -> str:
         return self._base_url + path
 
+    def _refresh_access_token(self):
+        """ Use the offline token to request a new access token """
+        self._headers.pop("Authorization", None)
+        # TODO: Make sure this is the right path
+        path = "/auth/token"
+        headers = {**self._headers}
+        response = self._request("POST", self._url(path), headers=headers, data={
+            'grant_type': 'refresh_token',
+            'refresh_token': self._token,
+            'client_id': 'public',
+        })
+        if response.status_code == 200:
+            self._headers["Authorization"] = "Bearer {}".format(response.cookies["access_token"])
```

Comment:
Is this the only place where the token is stored?

Originally, I was wondering if it made sense to store it locally (temporarily), but I suppose it doesn't matter and is cleaner this way.

Reply:
Yes the access token is only stored in the Authorization header. I decided against storing it independently since it isn't used anywhere else, is short lived, and a new one can be fetched if it is somehow lost.

Reply:
:+1: